### ISSUE TITLE
Dangling symlink traversal failure when using bud with layer cache

### ISSUE THREAD

**Issue by pkubatrh** [containers/buildah#1219]
**Description**
Image build fails when building an image using the layer cache from a Dockerfile while adding a symlink that will be dangling at build time due to the target not being located in the context directory.

**Steps to reproduce the issue:**
1.cat << EOF >Dockerfile
FROM fedora
COPY run-mysqld /tmp/
EOF
2. ln -s /bin/run-mysqld run-mysqld
3. buildah bud --layers .
4. buildah bud --layers .

**Describe the results you received:**
Second `buildah bud` fails with:
```
error checking if cached image exists from a previous build: error checking if copied files match: error resolving symlinks and comparing modified times: "/tmp/reproducer/run-mysqld": error checking if modified time of resolved symbolic link is greater: error walking file tree "/tmp/reproducer/run-mysqld": error evaluating symlink "run-mysqld": lstat /bin: no such file or directory
: exit status 1
```

**Describe the results you expected:**
Buildah sees that the symlink is dangling and does not try to check the target

**Output of `rpm -q buildah` or `apt list buildah`:**

buildah-1.5-2.gite94b4f9.fc28.x86_64

**Output of `buildah version`:**

Version:         1.5
Go Version:      go1.10.5
Image Spec:      1.0.0
Runtime Spec:    1.0.0
CNI Spec:        0.4.0
libcni Version:  
Git Commit:      
Built:           Thu Jan  1 01:00:00 1970
OS/Arch:         linux/amd64

**Comment by umohnani8** [comment]
@TomSweeneyRedHat no it wouldn't. It is failing because bud expects the symlink to exist so it can compare the timestamps to know whether to use from cache or not. Didn't think of a dangling symlink case. I guess the fix would be to ignore it if it is dangling and run the build process again i.e not use cached layers? Not sure what docker does.

**Comment by rhatdan** [comment]
@pkubatrh Does Docker handle this?

**Comment by rhatdan** [comment]
@umohnani8 @TomSweeneyRedHat @pkubatrh Is this still an issue or has it been fixed?

**Comment by pkubatrh** [comment]
@rhatdan Still seeing it in latest master

```
$ git rev-parse HEAD
36605c29eb258caa9720dbd42622269a81dd1bfa
$ cat << EOF >Dockerfile
FROM fedora
COPY test /tmp/
EOF
$ mkdir test; ln -s /bin/run-mysqld test/run-mysqld
$ ./buildah bud --layers .; ./buildah bud --layers .
...
STEP 1: FROM fedora
STEP 2: COPY test /tmp/
error checking if cached image exists from a previous build: error checking if copied files match: error resolving symlinks and comparing modified times: "/home/pkubat/go/buildah/src/github.com/containers/buildah/test": error checking if modified time of resolved symbolic link is greater: error walking file tree "/home/pkubat/go/buildah/src/github.com/containers/buildah/test": error evaluating symlink "test/run-mysqld": lstat /bin: no such file or directory
: exit status 1
```

**Comment by rhatdan** [comment]
@TomSweeneyRedHat is this the code you were working on for handling symlinks?

**Comment by TomSweeneyRedHat** [comment]
Yes and I need to revisit.

**Comment by umohnani8** [comment]
fixed in #1519
